astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -2219,6 +2233,7 @@ def __init__(
             end_col_offset=end_col_offset,
             parent=parent,
         )
+        self.doc_node = doc_node
```

Comment:
Same here with the assignment. In this particular case below `self.doc = doc` would make the most sense IMO.